File: 349.intersection-of-two-arrays.java
import java.util.HashMap;
import java.util.Map;

/*
 * @lc app=leetcode id=349 lang=java
 *
 * [349] Intersection of Two Arrays
 */

// @lc code=start
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int num : nums1) {
            map.put(num, map.getOrDefault(num, 0));
        }

        for (int num : nums2) {
            if (map.get(num) != null) {
                map.put(num, map.get(num) + 1);
            }

        }

        int cnt = 0;
        for (Map.Entry<Integer, Integer> i : map.entrySet()) {
            if (i.getValue() > 0)
                cnt++;
        }
        int[] res = new int[cnt];
        for (Map.Entry<Integer, Integer> i : map.entrySet()) {
            if (i.getValue() > 0) {
                res[--cnt] = i.getKey();
            }
        }
        return res;
    }
}
// @lc code=end

File: 349.intersection-of-two-arrays.py
#
# @lc app=leetcode id=349 lang=python3
#
# [349] Intersection of Two Arrays
#

# @lc code=start
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        # n1, n2
        # binary search? nlogn
        # hash map
        count1=Counter(nums1)
        count2=Counter(nums2)
        res=[]
        for num in count1:
            if num in count2:
                res.append(num)

        return res


# @lc code=end

